main_test.go

```diff
@@ -82,7 +82,7 @@ func TestEntrypoint(t *testing.T) {
 		}()
 		shutdown <- syscall.SIGTERM
 
-		time.Sleep(2 * time.Second)
+		time.Sleep(12 * time.Second)
```

Comment:
Let's substitute the boolean flag with a chan so that we can avoid sleeping all this time; also use `os.Setenv` to set the delay env var to a smaller delay